client/src/NetworkClient.ts

```diff
@@ -117,4 +209,47 @@ export class NetworkClient {
     public async removeTxFromMempool(txObj: PlainTransaction): Promise<void> {
         return this._eventClient.call('removeTxFromMempool', txObj);
     }
+
+    private _evictCachedTransactions() {
+        const CACHE_DURATION = 30;
+        // purge expired transactions
+        for (let i = 0; i < this._expiredTransactions.length; ++i) {
+            const [expiredAt] = this._expiredTransactions[i];
+            if (expiredAt + CACHE_DURATION <= this.headInfo.height) {
+                this._expiredTransactions.splice(i, 1);
+                --i;
+            }
+        }
+        // purge mined transactions
+        for (const tx of this._minedTransactions.values()) {
+            if (tx.blockHeight + CACHE_DURATION <= this.headInfo.height) {
+                this._minedTransactions.delete(tx.hash);
+            }
+        }
+        // purge relayed transactions
+        for (const tx of this._relayedTransactions.values()) {
+            if (tx.blockHeight! + CACHE_DURATION <= this.headInfo.height) {
+                this._relayedTransactions.delete(tx.hash!);
+            }
+        }
+    }
 }
+
+namespace NetworkClient { // tslint:disable-line:no-namespace
+    export enum Events {
+        API_READY = 'nimiq-api-ready',
+        API_FAIL = 'nimiq-api-fail',
+        CONSENSUS_SYNCING = 'nimiq-consensus-syncing',
+        CONSENSUS_ESTABLISHED = 'nimiq-consensus-established',
+        CONSENSUS_LOST = 'nimiq-consensus-lost',
+        PEERS_CHANGED = 'nimiq-peer-count',
+        BALANCES_CHANGED = 'nimiq-balances',
+        TRANSACTION_PENDING = 'nimiq-transaction-pending',
+        TRANSACTION_EXPIRED = 'nimiq-transaction-expired',
+        TRANSACTION_MINED = 'nimiq-transaction-mined',
+        TRANSACTION_RELAYED = 'nimiq-transaction-relayed',
+        HEAD_CHANGE = 'nimiq-head-change',
+    }
+}
+
+export { NetworkClient };
```

Comment:
How about exporting the events here, so that users of the NetworkClient can use them?

Reply:
The `Events` are accessible from outside via `NetworkClient.Events` also from outside

Reply:
Ah, because it's the class name. Ok, got it. :+1: